lib/utils/jsonPathUtils.ts

```diff
@@ -90,4 +90,65 @@ export class JsonPathUtils {
       }
     }
   }
+
+  static changeSpecialPathsRecursively(pd: IPresentationDefinition) {
+    const paths: { value: unknown; path: (string | number)[] }[] = JsonPathUtils.extractInputField(pd, ['$..path']);
+    for (const path of paths) {
+      this.modifyPathsWithSpecialCharacter(path.path, pd);
+    }
+  }
+
+  private static modifyPathsWithSpecialCharacter(pathDetails: (string | number)[], pd: IPresentationDefinition) {
+    let objectCursor: any = pd;
+    for (let i = 1; i < pathDetails.length; i++) {
+      if (i + 1 < pathDetails.length) {
+        objectCursor = objectCursor[pathDetails[i]];
+      }
+      if (pathDetails.length == i + 1) {
+        const paths: string[] = objectCursor[pathDetails[i]];
+        const editedPaths: string[] = [];
+        for (let j = 0; j < paths.length; j++) {
+          editedPaths.push(this.modifyPathWithSpecialCharacter(paths[j]));
+        }
+        objectCursor[pathDetails[i]] = editedPaths;
+        break;
+      }
+    }
+  }
+
+  private static modifyPathWithSpecialCharacter(path: string): string {
+    const REGEX_PATH_ESCAPED = /\['@\w+']/g;
+    const REGEX_PATH = /@\w+/g;
```

Comment:
It seems you are using 2 regexes to remove the ['@jsonld'] case. Why not use a lookahead and look behind in the regex? That means it will only match when it is not already in this form.

Something like this should work in one go:
```
/(?<!\[')@\w+(?!'])/g
````
What it does, is check whether @ + word matches. If that is the case it does a negative look behind (?<!) for the prefix [' and then a negative look ahead (?!) for the suffix ']

Reply:
Btw you might need to introduce a capturing group for @\w+ to get only that value, depending on the rest of the implementation